src/utils/hooks/useAnchor/useAnchor.tsx

```diff
@@ -38,10 +39,21 @@ export function useAnchor(
 
   useIsomorphicLayoutEffect(() => {
     if (ref.current) {
-      const rectRef: DOMRect = ref.current.getBoundingClientRect();
-      const { width, height, top, left } = rectRef;
+      const updateRect = () => {
+        const rectRef: DOMRect = ref.current.getBoundingClientRect();
+        const { width, height, top, left } = rectRef;
 
-      if (rect.top !== top) rectSet({ width, height, top, left });
+        if (rect.top !== top) rectSet({ width, height, top, left });
+      };
+
+      window.addEventListener(
+        'resize',
+        throttle(() => {
+          updateRect();
+        }, 10)
+      );
```

Comment:
The event listener callback needs to be declared separately and the listener removed in the effect cleanup.